src/threads/thread_pool/mod.rs

```diff
@@ -342,6 +348,40 @@ impl<F: RangeFactory> ThreadPoolImpl<F> {
                                 }
                             }
                         }
+                        #[cfg(all(not(miri), target_os = "windows"))]
+                        match cpu_pinning {
+                            CpuPinningPolicy::No => (),
+                            // SAFETY: `GetCurrentThread()` always returns a valid handle for the
+                            // current thread if it's used within the same thread.
+                            //
+                            // `SetThreadAffinityMask()` may fail if the requested CPU isn't in
+                            // the current process's affinity mask or if the affinity mask can't be
+                            // used for another reason. In these cases it will fail gracefully.
+                            //
+                            // `GetLastError()`is used when an error is returned from
+                            // `SetThreadAffinityMask()`, and should be safe to call in this
+                            // context.
+                            CpuPinningPolicy::IfSupported => {
+                                let affinity_mask = 1usize << id;
+                                let result = unsafe { SetThreadAffinityMask(GetCurrentThread(), affinity_mask) };
+                                if result == 0 {
+                                    let _last_error = unsafe { GetLastError() };
```

Comment:
I'm surprised Clippy didn't complain about having a SAFETY comment just before this `unsafe {}` block. It looks like the `lints.yml` workflow also needs an OS matrix.